app/controllers/tests_controller.rb

```diff
@@ -85,19 +94,27 @@ def destroy
     end
   end
 
-
-
   private
 
   def set_assignment
     @assignment = Assignment.find(params[:assignment_id])
   end
   # Use callbacks to share common setup or constraints between actions.
+  def set_test_grouping
```

Comment:
Might be missing it, but where is the `set_test_grouping` private function being called in the other controller functions?

Reply:
I am not sure, in the test controller there is line   # before_action :set_test_grouping in line 6. I think originally we want to build a nested structure, that first build the test group and then build a test case. I this code version this part of code can be commented to improve the coverage I guess.

Reply:
@yaoya2618-dot just remove it if it is unused

Reply:
removed